modules/integration/tests-integration/tests-patches/src/test/java/org/wso2/carbon/esb/nhttp/transport/json/test/ESBJAVA4940CharacterEncodingRemovalTestCase.java

```diff
@@ -0,0 +1,125 @@
+        serverConfigurationManager.applyConfiguration(
+                Paths.get(getESBResourceLocation(), "nhttp", "transport", "ESBJAVA4940", "axis2.xml").toFile());
+
+        // Enable wire logs
+        String carbonHome = System.getProperty(ServerConstants.CARBON_HOME);
+        File log4jProperties = Paths.get(carbonHome, "repository", "conf", "log4j.properties").toFile();
+        applyProperty(log4jProperties, "log4j.logger.org.apache.synapse.transport.http.wire", "DEBUG");
+        serverConfigurationManager.restartGracefully();
+        super.init();
+
+        loadESBConfigurationFromClasspath("/artifacts/ESB/nhttp/transport/ESBJAVA4940/synapseConfig.xml");
+
+        logViewer = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
+    }
+
+
+    @Test(groups = "wso2.esb",
+          description = "Test charset encoding removal when SetEncoding is true",
+          enabled = true)
+    public void testRemoveCharsetSetEncodingPropertyTrue() throws Exception {
+        Assert.assertTrue(isCharsetEncodingPresent(getProxyServiceURLHttp("ESBJAVA4940SetEncodingTrue")),
+                "Charset Encoding is not present in the request sent");
+    }
+
+    @Test(groups = "wso2.esb",
+          description = "Test charset encoding removal when SetEncoding is false",
+          enabled = true,
+          dependsOnMethods = "testRemoveCharsetSetEncodingPropertyTrue")
+    public void testRemoveCharsetSetEncodingPropertyFalse() throws Exception {
+        Assert.assertFalse(isCharsetEncodingPresent("ESBJAVA4940SetEncodingFalse"),
+                "Charset Encoding is present in the request sent");
+    }
+
+    private boolean isCharsetEncodingPresent(String proxyURL) throws Exception {
+        logViewer.clearLogs();
+        String payload = "{\"sampleJson\" : \"sampleValue\"}";
+        Map<String, String> requestHeader = new HashMap<>();
+        requestHeader.put("Content-Type", "application/json");
+        try {
+            HttpRequestUtil.doPost(new URL(getProxyServiceURLHttp(proxyURL)), payload, requestHeader);
+            Thread.sleep(30000);
+        } catch (Exception e) {
+            //Ignored
+        }
+
+        LogEvent[] logs = logViewer.getAllRemoteSystemLogs();
+        for (LogEvent log : logs) {
+            if (log.getMessage().contains("Content-Type: application/json; charset=UTF-8")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void applyProperty(File srcFile, String key, String value) throws Exception {
+        File destinationFile = new File(srcFile.getName());
+        Properties properties = new Properties();
+        properties.load(new FileInputStream(srcFile));
+        properties.setProperty(key, value);
+        properties.store(new FileOutputStream(destinationFile), null);
+        serverConfigurationManager.applyConfigurationWithoutRestart(destinationFile);
+    }
+
+    @AfterClass
+    public void cleanUp() throws Exception {
```

Comment:
Need to restore configurations using serverConfigurationManager.restoreToLastConfiguration()